File: Shared/Blog.Contracts/V1/Requests/CategoriesRequests/UpdateCategoryRequest.cs
namespace Blog.Contracts.V1.Requests.CategoriesRequests;

using System.ComponentModel.DataAnnotations;
using Interfaces;

/// <summary>
/// Update category request.
/// </summary>
public class UpdateCategoryRequest : IRequest
{
    /// <summary>
    /// Gets or sets the parent category id.
    /// </summary>
    public int? ParentCategoryId { get; set; }

    /// <summary>
    /// Gets or sets the name.
    /// </summary>
    /// <value>
    /// The name.
    /// </value>
    [Required]
    public string Name { get; set; }

    /// <summary>
    /// Gets or sets the description.
    /// </summary>
    /// <value>
    /// The description.
    /// </value>
    public string Description { get; set; }
}

File: Shared/Blog.FSharp.Contracts/V1/Requests/CategoriesRequests/UpdateCategoryRequest.fs
namespace Blog.FSharp.Contracts.V1.Requests.CategoriesRequests

open System.ComponentModel.DataAnnotations
open Blog.FSharp.Contracts.V1.Requests.Interfaces

/// <summary>
/// Update category request.
/// </summary>
type UpdateCategoryRequest() =
    interface IRequest

    /// <summary>
    /// Gets or sets the parent category id.
    /// </summary>
    member val ParentCategoryId: int option = None with get, set

    /// <summary>
    /// Gets or sets the name.
    /// </summary>
    [<Required>]
    member val Name: string = "" with get, set

    /// <summary>
    /// Gets or sets the description.
    /// </summary>
    member val Description: string = "" with get, set